app.py

```diff
@@ -38,34 +44,46 @@
     }
 
     config = None
-
-    with open('config.yml', 'r') as config_file:
-        try:
-            config = yaml.load(config_file)
-        except yaml.YAMLError as err:
-            print(err)
-            exit(1)
-
+    config_path = None
+    if args.config is not None:
+        config_path = args.config
+    else:
+        config_path = 'config.yml'
+
+    try: 
+        with open(config_path, 'r') as config_file:
+            try:
+                config = yaml.load(config_file)
+            except yaml.YAMLError as err:
+                print(err,file=sys.stderr)
+                exit(1)
+    except FileNotFoundError as err:
+        print('No such config file', file=sys.stderr)
+        exit(1)
+    
     if config is None:
-        print('Error with your config')
+        print('Error with your config', file=sys.stderr)
         exit(1)
-    elif 'pb_token' not in config:
-        print('Config must contain your Pushbullet Access Token!')
+    elif use_pushbullet == True and 'pb_token' not in config:
+        print('Config must contain your Pushbullet Access Token!', file=sys.stderr)
         exit(1)
     elif ('term' not in config) or (config['term'] not in terms):
-        print('No term (or an incorrect term) specified in config!')
+        print('No term (or an incorrect term) specified in config!', file=sys.stderr)
         exit(1)
     elif 'year' not in config:
-        print('Year not specified in config!')
+        print('Year not specified in config!', file=sys.stderr)
         exit(1)
 
-    pb = Pushbullet(config['pb_token'])
+    if use_pushbullet is True:
+        from pushbullet import Pushbullet
+        pb = Pushbullet(config['pb_token'])
+
     semester = str(config['year'])+terms.get(config['term'])
 
     last_open_courses = list()
 
     while True:
-        print("Starting check... ", datetime.now().strftime('%Y-%m-%d %H:%M:%S'))
+        print("Starting check... ", datetime.now().strftime('%Y-%m-%d %H:%M:%S'), file=sys.stderr)
```

Comment:
Probably should print to stdout instead since it's not an error. If this is so it prints out right away, the stdout buffer can be flushed with sys.stdout.flush()

Reply:
I was thinking since it is largely diagnostic verbosity rather than the actual results that are wanted, stderr makes more sense; makes it easier to parse the actual results in a pipeline.